### PR TITLE
feat(NgTemplateOutlet): add context to NgTemplateOutlet

### PR DESCRIPTION
**Pull request by shlomiassaf**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [?] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

SEE #8368 

**Does this PR introduce a breaking change?**
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by shlomiassaf** [comment]
@fxck @pkozlowski-opensource

**Comment by fxck** [comment]
tested on my app as a solution to both https://github.com/angular/angular/issues/9033 and http://stackoverflow.com/questions/37654004/angular2-feeding-data-back-to-template-from-ngtemplateoutlet works great

**Comment by alxhub** [comment]
LGTM. @IgorMinar, can you review the public API change?

File: modules/@angular/common/src/directives/ng_template_outlet.ts [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
  * Creates and inserts an embedded view based on a prepared `TemplateRef`.
+ * You can attach a context object to the `EmbeddedViewRef` by setting `[ngOutletContext]`.
+ * `[ngOutletContext]` should be an object, the object's keys will be the local template variables
+ * available within the `TemplateRef`.
+ *
+ * Note: using the key `$implicit` in the context object will set it's value as default.
  *
  * ### Syntax
- * - `<template [ngTemplateOutlet]="templateRefExpression"></template>`
+ * - `<template [ngTemplateOutlet]="templateRefExpression" [ngOutletContext]="ObjectExpression"></template>`
```
Comment (pkozlowski-opensource) [inline modules/@angular/common/src/directives/ng_template_outlet.ts]: `ObjectExpression ` -> `objectExpression ` - lower-case `O`

Comment (shlomiassaf) [inline modules/@angular/common/src/directives/ng_template_outlet.ts]: Done

**Pull request closed without merging**